azterraform Dockerfile: golang installation hardcodes amd64 architecture

The amd64 ARCH for golang is hardcoded:

https://github.com/Azure/tfmod-scaffold/blob/bcf0e17c604733d179d1ec2c9fc2a224ebdc36bb/Dockerfile#L38-L40

Because of this on Apple Silicon the container fails in this way:

```
% docker run --rm -v $(pwd):/src -w /src mcr.microsoft.com/azterraform:v0.8.0 make pre-commit
==> Formatting terraform code...
terraform fmt -recursive
==> Fixing test and document terraform blocks code with terrafmt...
qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory
qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory
make: *** [tfmod-scaffold/GNUmakefile:66: depsensure] Error 255
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -35,9 +35,9 @@ ENV PATH=$PATH:/usr/local/go/bin
 COPY --from=build /go/bin /usr/local/go/bin
 COPY .terraformrc /root/.terraformrc
 RUN yum update && yum install -y yum ca-certificates zip unzip jq nodejs python3-pip make git diffutils build-essential && \
-    wget https://go.dev/dl/go${GOLANG_IMAGE_TAG}.linux-amd64.tar.gz && \
-    tar -C /usr/local -xzf go*.linux-amd64.tar.gz && \
-    rm go${GOLANG_IMAGE_TAG}.linux-amd64.tar.gz && \
+    wget https://go.dev/dl/go${GOLANG_IMAGE_TAG}.linux-${BUILDARCH}.tar.gz && \
+    tar -C /usr/local -xzf go*.linux-${BUILDARCH}.tar.gz && \
+    rm go${GOLANG_IMAGE_TAG}.linux-${BUILDARCH}.tar.gz && \
     npm install markdown-table-formatter -g && \
     mkdir -p $HOME/.terraform.d/plugin-cache
 RUN pip3 install --upgrade setuptools && \

